presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -112,13 +112,21 @@ public CachingHiveMetastore(@ForCachingHiveMetastore HiveMetastore delegate, @Fo
                 config.getMetastoreCacheMaximumSize());
     }
 
-    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Duration refreshInterval, long maximumSize)
+    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
     {
         this(
                 delegate,
                 executor,
                 OptionalLong.of(cacheTtl.toMillis()),
-                refreshInterval.toMillis() >= cacheTtl.toMillis() ? OptionalLong.empty() : OptionalLong.of(refreshInterval.toMillis()),
+                refreshInterval
+                        .flatMap(interval -> {
+                            if (interval.toMillis() >= cacheTtl.toMillis()) {
+                                return Optional.empty();
```

Comment:
does it mean that for `refreshInterval` empty means no refresh (differently than for `expiresAfterWriteMillis`).

Reply:
- empty `refreshInterval` means no refresh
- empty `expiresAfterWriteMillis` means no expiry

so this is consistent